### ISSUE TITLE
feat: response interceptor

### ISSUE THREAD

**Issue by chimurai** [chimurai/http-proxy-middleware#515]
Add `responseInterceptor` handler to easily intercept and manipulate responses. Responses compressed with `brotli`, `gzip` and `deflate` are decompressed automatically.

install beta: `npm install http-proxy-middleware@1.2.0-beta.2` or `yarn add http-proxy-middleware@1.2.0-beta.2`
 
- fixes #97
- fixes #469
- fixes #355

TL;DR

```javascript
const { createProxyMiddleware, responseInterceptor } = require('http-proxy-middleware');

const proxy = createProxyMiddleware({
  /**
   * IMPORTANT: avoid res.end being called automatically
   **/
  selfHandleResponse: true, // res.end() will be called internally by responseInterceptor()

  /**
   * Intercept response and replace 'Hello' with 'Goodbye'
   **/
  onProxyRes: responseInterceptor(async (responseBuffer, proxyRes, req, res) => {
    const response = responseBuffer.toString('utf-8'); // convert buffer to string
    return response.replace('Hello', 'Goodbye'); // manipulate response and return the result
  }),
});

```

documentation: https://github.com/chimurai/http-proxy-middleware/tree/response-interceptor#intercept-and-manipulate-responses

recipes: https://github.com/chimurai/http-proxy-middleware/blob/response-interceptor/recipes/response-interceptor.md#readme

**Comment by chimurai** [comment]
published in `v1.2.0`

**Comment by OriginalEXE** [comment]
Just wanted to say thank you for implementing this feature, you are awesome.
